mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,193 @@
+            context.uid,
+            context.project,
+        )
+        env["MLFLOW_EXPERIMENT_ID"] = experiment_id
+        self.mlflow_experiment = experiment_id
+        return env
+
+    def _apply_post_run_tasks(
+        self,
+        context: Union[MLClientCtx, dict],
+    ):
+        """
+        Performs post-run tasks of logging 3rd party artifacts generated during the run.
+        :param context: current mlrun context
+        """
+        experiment_id = self.mlflow_experiment
+        runs = self._client.search_runs(
+            experiment_id, filter_string=f'tags.mlrun_runid="{context.uid}"'
+        )
+        if not runs:
+            experiments = [
+                experiment.experiment_id
+                for experiment in self._client.search_experiments()
+            ]
+            runs = self._client.search_runs(
+                experiments, filter_string=f'tags.mlrun_runid="{context.uid}"'
+            )
+
+        if runs:
+            for run in runs:  # each run gets logged separately
+                self._log_run(context, run)
+
+    def _log_run(self, context: MLClientCtx, run):
+        """
+        after mlrun function's run is done, copy all data logged by third party app tracker
+        :param context: current mlrun context
+        :param run: mlflow run to log
+        """
+        model_paths = []
+        logger.debug("Starting to log MLFlow params")
+        for key, val in run.data.params.items():
+            context._parameters[key] = val
+        logger.debug("Finished to log MLFlow params")
+        context.log_results(run.data.metrics)
+        context.set_label("mlflow-runid", run.info.run_id)
+        context.set_label("mlflow-experiment", run.info.experiment_id)
+        logger.debug("Starting to log MLFlow artifacts")
+        for artifact in self._client.list_artifacts(run.info.run_id):
+            full_path = self._tracked_platform.artifacts.download_artifacts(
+                run_id=run.info.run_id, artifact_path=artifact.path
+            )
+            if artifact.is_dir and os.path.exists(os.path.join(full_path, "MLmodel")):
+                model_paths.append(
+                    full_path
+                )  # this is the model folder, we log it after logging all artifacts
+            else:
+                self.log_artifact(context, full_path, artifact)
+        logger.debug("Finished to log MLFlow artifacts")
+        if model_paths:
+            logger.debug("Starting to log MLFlow model")
+            for path in model_paths:
+                self.log_model(path, context)
+            logger.debug("Finished to log MLFlow model")
+
+    def log_model(
+        self,
+        model_uri: str,
+        context: Union[MLClientCtx, dict],
+    ):
+
+        model_info = self._tracked_platform.models.get_model_info(model_uri=model_uri)
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            model_zip = f"{tmp_dir}-model.zip"
+        zip_folder(model_uri, model_zip)
```

Comment:
The rest of the function need to also be within the context manager scope, otherwise at this point (when calling `zip_folder`) the temp directory was already removed... 
You basically need to put the rest of the function within the `with` scope.